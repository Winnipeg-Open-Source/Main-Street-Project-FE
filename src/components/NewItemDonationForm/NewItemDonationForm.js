import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Select } from 'pcln-design-system';
import Form from 'components/Form';
import FoodDonationForm from 'components/FoodDonationForm';
import ClothesDonationForm from 'components/ClothesDonationForm';
import HouseholdItemDonationForm from 'components/HouseholdItemDonationForm';
import MiscellaneousDonationForm from 'components/MiscellaneousDonationForm';

function FormSwitch ({ itemType, ...props }) {
    switch (itemType) {
        case 'Food':
            return <FoodDonationForm {...props} />;
        case 'Clothing':
            return <ClothesDonationForm {...props} />;
        case 'Household Item':
            return <HouseholdItemDonationForm {...props} />;
        case 'Miscellaneous':
            return <MiscellaneousDonationForm {...props} />;
    }

    return null;
}

function NewItemDonationForm ({ className, onCancelClick, onSaveClick, ...props }) {
    const [ itemType, setItemType ] = useState();

    const onItemTypeChange = (evt) => setItemType(evt.target.value);

    return (
        <Form className={className} title='New Item' onCancelClick={onCancelClick} onSaveClick={onSaveClick}>
            <Select id='item-type-select' value={itemType} onChange={onItemTypeChange}>
                <option disabled selected>Item Type</option>
                <option>Food</option>
                <option>Clothing</option>
                <option>Household Item</option>
                <option>Miscellaneous</option>
            </Select>
            <FormSwitch itemType={itemType} {...props} />
        </Form>
    );
}

NewItemDonationForm.displayName = 'NewItemDonationForm';

NewItemDonationForm.propTypes = {
    className: PropTypes.string,
    onCancelClick: PropTypes.func,
    onSaveClick: PropTypes.func,
};

NewItemDonationForm.defaultProps = {
    className: '',
};

export default NewItemDonationForm;
