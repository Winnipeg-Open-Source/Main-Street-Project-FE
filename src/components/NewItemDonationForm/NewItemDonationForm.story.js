import React from 'react';
import { action } from '@storybook/addon-actions';
import NewItemDonationForm from './NewItemDonationForm';

const genericProps = {
    onCancelClick: action('cancel clicked'),
    onSaveClick: action('save clicked'),
};

const props = {
    name: 'Item Name',
    description: 'Item Description',
    quantity: 5,
    size: 'L',
    retailValue: 500,
    condition: 'Great',
    notes: 'Some notes',
    onNameChange: action('name changed'),
    onDescriptionChange: action('description changed'),
    onQuantityChange: action('quantity changed'),
    onSizeChange: action('size changed'),
    onRetailValueChange: action('retail value changed'),
    onConditionChange: action('condition changed'),
    onNotesChange: action('notes changed'),
};

export const EmptyForm = () => <NewItemDonationForm {...genericProps} />;

export const CompletedForm = () => <NewItemDonationForm {...genericProps} {...props} />;

export default {
    title: 'NewItemDonationForm',
    component: NewItemDonationForm,
};
