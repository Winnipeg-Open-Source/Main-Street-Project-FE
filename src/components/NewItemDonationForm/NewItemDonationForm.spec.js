import React from 'react';
import { render } from '@testing-library/react';
import NewItemDonationForm from './NewItemDonationForm';

describe ('NewItemDonationForm', () => {
    it ('renders without crashing', () => {
        const { asFragment } = render(<NewItemDonationForm />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('can fill out food donation form', () => {

    });

    it ('can fill out clothing donation form', () => {

    });

    it ('can fill out household item donation form', () => {

    });

    it ('can fill out miscellaneous donation form', () => {

    });
});
