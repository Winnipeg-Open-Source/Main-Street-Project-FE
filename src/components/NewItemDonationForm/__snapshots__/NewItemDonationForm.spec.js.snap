// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewItemDonationForm renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pbJYR hNzbLI sc-pkfPC jpeBiC"
  >
    <div
      class="sc-AxjAm sc-AxiKw sc-qQxFs cDVLUs"
    >
      <div
        class="sc-AxhCb wyOpT"
      >
        New Item
      </div>
      <div
        class="sc-AxjAm sc-AxiKw sc-qYUWV foBumL"
      >
        <div
          class="sc-AxjAm sc-AxiKw iSVrPI"
          width="1"
        >
          <select
            class="sc-pZCuu kQVKTB sc-oTaid fgxYOF"
            data-testid="item-type-select"
            font-size="2,,1"
            id="item-type-select"
          >
            <option
              disabled=""
            >
              Item Type
            </option>
            <option>
              Food
            </option>
            <option>
              Clothing
            </option>
            <option>
              Household Item
            </option>
            <option>
              Miscellaneous
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="sc-Axmtr gXMRoh sc-fzqPZZ sc-pRFHb JnCiE"
            color="text.light"
            fill="currentcolor"
            focusable="false"
            height="24"
            role="img"
            tabindex="-1"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-AxjAm sc-AxiKw sc-psDhf dCBSxT"
      />
    </div>
  </div>
</DocumentFragment>
`;
